platform/mock-local-home-platform.ts

```diff
@@ -69,38 +108,56 @@ export class MockLocalHomePlatform implements MockUDPListener {
 
   private sendUDPBroadcast(scanConfig: UDPScanConfig) {
     const packetBuffer = Buffer.from(scanConfig.discoveryPacket, 'hex');
-    this.network.sendUDPMessage(
+    this.mockNetwork.sendUDPMessage(
       packetBuffer,
       scanConfig.broadcastPort,
       scanConfig.broadcastAddress,
       scanConfig.listenPort,
       scanConfig.broadcastAddress
     );
   }
+  private isPromise<RS>(response: RS | Promise<RS>): response is Promise<RS> {
+    return (response as Promise<RS>).then !== undefined;
+  }
+
+  private processIntentResponse<RS>(response: RS | Promise<RS>): Promise<RS> {
+    if (this.isPromise<RS>(response)) {
+      return response;
+    } else {
+      return Promise.resolve(response);
+    }
+  }
 
   // Establish fulfillment path using app code
   onUDPMessage(msg: Buffer, rinfo: RemoteAddressInfo): void {
     console.log('received discovery payload:', msg, 'from:', rinfo);
-    const identifyRequest: string = JSON.stringify({
+
+    const identifyRequest: smarthome.IntentFlow.IdentifyRequest = {
       requestId: 'request-id',
       inputs: [
         {
-          intent: 'action.devices.IDENTIFY',
+          intent: smarthome.Intents.IDENTIFY,
           payload: {
             device: {
               radioTypes: [],
-              udpScanData: msg.toString('hex'),
+              udpScanData: { data: msg.toString('hex') },
             },
             structureData: {},
             params: {},
           },
         },
       ],
       devices: [],
-      rinfo,
+    };
+
+    this.processIntentResponse<smarthome.IntentFlow.IdentifyResponse>(
```

Comment:
wouldn't this work?
```
Promise.resolve(this.app.identifyHandler(identifyRequest)).then(...)
```
If `identifyHandler` result is a Promise, it will cascade and if it's a result it will wrap it in a promise.

Reply:
That makes a lot more sense.  Will make this change

Reply:
If you mark the function as `async` you can `await` and make the code look more synchronous.

Reply:
Neat. Done.